### PR TITLE
ENG-1201: Add eslint support

### PR DESCRIPTION
**Pull request by timmyjose**
Changes:
  - added eslint support.
  - added eslint rules from the `explorer` repository.
  - source files modified as per `npm run fix`.
  - skipping `component/pkg` directory from eslint rules.
  - updated `_app.tsx` as per the rules.

### CODE REVIEW

**Comment by calummoore** [comment]
I'm not sure why, but the Home.tsx and Why.tsx both reformat on save, but are not picked up by the eslint fix command. Is that the same for you?

**Comment by timmyjose** [comment]
> I'm not sure why, but the Home.tsx and Why.tsx both reformat on save, but are not picked up by the eslint fix command. Is that the same for you?

Hmmm, that is strange. I'll have a look and see.

**Comment by timmyjose** [comment]
> > I'm not sure why, but the Home.tsx and Why.tsx both reformat on save, but are not picked up by the eslint fix command. Is that the same for you?
> 
> Hmmm, that is strange. I'll have a look and see.

@calummoore I just tried it locally on this branch, and it seems to work for me. The behaviour:
   * If I change the indentation in VSCode, I get the error: "Expected indentation of 2, but found ..."
   * Upon saving, it iutomatically indents to 2 spaces.
   * In Vim (where I don't have auto-format for TS), I can save it with some arbitrary indentation, say 6, and when I run 
     `lint`, I get (for `Home.tsx`):

      ```
       11:1  error  Expected indentation of 2 spaces but found 6  @typescript-eslint/indent
      ```
  * If I then run `npm run fix`, it resets indentation to 2 spaces, and a subsequent run of `npm run lint` yields no errors.

**Comment by timmyjose** [comment]
Updated to use `yarn` throughout (in lieu of `pnpm`) as per https://github.com/polybase/polylang/pull/131#discussion_r1349760388

File: site/components/Why.tsx [hunk 1]
```diff
@@ -1,42 +1,42 @@
...
-        <Stack spacing='10'>
-            <Heading as='h2'>Why?</Heading>
-            <SimpleGrid columns={1} spacing={10} minChildWidth='300px'>
-                <WhyBlock title='Verifiable computation'>
+  return (
+    <Stack spacing='10'>
+      <Heading as='h2'>Why?</Heading>
+      <SimpleGrid columns={1} spacing={10} minChildWidth='300px'>
+        <WhyBlock title='Verifiable computation'>
                     Run computations and generate a proof that the computation was run as designed.
```
Comment (calummoore) [inline site/components/Why.tsx:20]: My editor is still re-formatting this line?

File: site/components/Home.tsx [hunk 2]
```diff
@@ -1,46 +1,46 @@
...
+    <Box maxW='container.lg' margin='0 auto'>
+      <ColorModeSync />
+      <Stack spacing='6em'>
+        <Box py={3}>
+          <Navbar />
+        </Box>
+        <Stack spacing='10em'>
+          <Stack spacing={6}>
+            <Heading as='h1' fontSize={['3em', '4em']}>
                             TypeScript for<br /> Zero Knowledge
```
Comment (calummoore) [inline site/components/Home.tsx:21]: And this line

Comment (timmyjose) [reply site/components/Home.tsx:21]: I'll update the changes from `main` and double-check on this.

File: site/components/Why.tsx (hunk 1 above)
Comment (timmyjose) [reply site/components/Why.tsx:20]: @calum I checked on my local copy of `explorerer` (from where the `eslint` rules were sourced), and here's what I found (note that I'm not very familiar with VSCode) which mirrors the behaviour of this PR:
  1. Without the `eslintConfig` section (or `eslint` in general), VSCode reformats on save (default behaviour, I presume since I'd not tweaked those on my VSCode instance).
  2. With the `eslintConfig` rules present, VSCode reformats on save again, but respects the `indent` parameter passed in (2, 4, etc.) **but** `eslint` does not mark the text nodes/entries as errors based on the indentation.
  3. So even if we were to manually set indents to some random spacing using some other plain text editor, `eslint` would not recognise that indentation as erroneous (and so `fix` would do nothing as well). However, when saving in VSCode, it would still auto-format the text nodes as well (according to the value of the "Spaces: ..." value shown in the bottom-right of the editor window).

tl;dr: So `eslint` by itself seems to be working correctly (not recognising text nodes), and I couldn't find any configuration for `eslint` relating to this.

VSCode seems to auto-format on save by default and it does so even for text nodes.

**Review by calummoore — APPROVED** [review]
